operator/v1/tests/machineconfigurations.operator.openshift.io/IrreconcilableMachineConfig.yaml

```diff
@@ -0,0 +1,116 @@
+          operatorLogLevel: Normal
+          irreconcilableValidationOverrides: {}
+    - name: Should be able to create a IrreconcilableValidationOverrides configuration knob that enables all overrides
+      initial: |
+        apiVersion: operator.openshift.io/v1
+        kind: MachineConfiguration
+        spec:
+          irreconcilableValidationOverrides:
+            storage:
+              - Disks
+              - FileSystems
+              - Raid
+      expected: |
+        apiVersion: operator.openshift.io/v1
+        kind: MachineConfiguration
+        spec:
+          logLevel: Normal
+          operatorLogLevel: Normal
+          irreconcilableValidationOverrides:
+            storage:
+              - Disks
+              - FileSystems
+              - Raid
+    - name: Should be able to create a IrreconcilableValidationOverrides configuration knob that only enables a set of overrides
+      initial: |
+        apiVersion: operator.openshift.io/v1
+        kind: MachineConfiguration
+        spec:
+          irreconcilableValidationOverrides:
+            storage:
+              - Disks
+              - FileSystems
+      expected: |
+        apiVersion: operator.openshift.io/v1
+        kind: MachineConfiguration
+        spec:
+          logLevel: Normal
+          operatorLogLevel: Normal
+          irreconcilableValidationOverrides:
+            storage:
+              - Disks
+              - FileSystems
+    - name: Should not be able to create a IrreconcilableValidationOverrides with invalid storage override values
+      initial: |
+        apiVersion: operator.openshift.io/v1
+        kind: MachineConfiguration
+        spec:
+          irreconcilableValidationOverrides:
+            storage:
+              - Disks
+              - invalid
+      expectedError: "spec.irreconcilableValidationOverrides.storage[1]: Unsupported value: \"invalid\": supported values: \"Disks\", \"FileSystems\", \"Raid\", <nil>"
+
+    - name: Should not be able to set set duplicated elements in the storage section of IrreconcilableValidationOverrides.
+      initial: |
+        apiVersion: operator.openshift.io/v1
+        kind: MachineConfiguration
+        spec:
+          irreconcilableValidationOverrides:
+            storage:
+              - Disks
+              - FileSystems
+              - Disks
+      expectedError: "spec.irreconcilableValidationOverrides.storage[2]: Duplicate value: \"Disks\""
+  onUpdate:
+    - name: Should be able to remove the irreconcilableValidationOverrides and restore the default behavior.
+      initial: |
+        apiVersion: operator.openshift.io/v1
+        kind: MachineConfiguration
+        spec:
+          irreconcilableValidationOverrides:
+            storage:
+              - Disks
+              - FileSystems
+              - Raid
+      updated: |
+        apiVersion: operator.openshift.io/v1
+        kind: MachineConfiguration
+        spec:
+          irreconcilableValidationOverrides: {}
```

Comment:
This shouldn't be permitted, make sure to update this and add the min properties